### PR TITLE
Add FromMnemonic, change namescope and add github workflow file

### PR DESCRIPTION
**Pull request by paweljakubas**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        plan:
+        - { build: stack, resolver: "--resolver lts-14.25"  }
+        include: []
+
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Install OS Packages
+      uses: mstksg/get-package@v1
+      with:
+        apt-get: libgmp-dev libssl-dev libtinfo-dev libsystemd-dev zlib1g-dev
```
Comment (KtorZ) [inline .github/workflows/main.yml:22]: small note: I need libssl-dev and libsystemd-dev (and probably most of the other except libgmp-dev) for building the consensus and network libraries in cardano-transactions.
I think that here, only `libgmp-dev` and `libssl-dev` are necessary for building cryptonite & cardano-crypto.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    - uses: actions/checkout@v1
+
+    - name: Setup Stack
+      uses: mstksg/setup-stack@v1
+
+    - name: Build & Test
+      run: |
+        stack --no-terminal test --bench --only-dependencies
+        stack --no-terminal test --bench --no-run-benchmarks
```
Comment (KtorZ) [inline .github/workflows/main.yml:32]: I'll suggest also setting up cache steps here.

File: src/Cardano/Mnemonic.hs [hunk 3]
```diff
@@ -263,3 +280,95 @@ mnemonicToText =
...
+    (SomeMnemonic mwa) == (SomeMnemonic mwb) =
+        case typeOf mwa `testEquality` typeOf mwb of
+            Nothing -> False
+            Just Refl -> mwa == mwb
+
+-- | Create a passphrase from a mnemonic sentence. This class enables caller to
+-- parse text list of variable length into mnemonic sentences.
+--
+-- >>> fromMnemonic @'[12,15,18,21] @"generation" ["toilet", "curse", ... ]
+-- Right (Passphrase <ScrubbedBytes>)
```
Comment (KtorZ) [inline src/Cardano/Mnemonic.hs:298]: That comment is actually outdated. `FromMnemonic` used to return a `Passphrase` but it now returns `SomeMnemonic`.

File: .github/workflows/main.yml (hunk 2 above)
Comment (paweljakubas) [reply .github/workflows/main.yml:42]: indeed, added caching and haddoc

File: src/Cardano/Mnemonic.hs (hunk 3 above)
Comment (paweljakubas) [reply src/Cardano/Mnemonic.hs]: corrected

File: .github/workflows/main.yml (hunk 1 above)
Comment (paweljakubas) [reply .github/workflows/main.yml]: yes, `libgmp-dev` and `libssl-dev` are enough

**Pull request merged**